### PR TITLE
DaedalusIPC clean ups

### PR DESCRIPTION
**Pull request by rvl**
For #209.

I would also like to add the javascript test case from cardano-wallet, but am out of time today.

### CODE REVIEW

File: app/daedalus-ipc.hs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+module Main where
+
+import           Cardano.Prelude
+
+import           Cardano.BM.Configuration.Static (defaultConfigStdout)
+import           Cardano.BM.Setup (setupTrace_)
+import           Cardano.Shell.DaedalusIPC
+
+main :: IO ()
+main = fmap readEither <$> getArgs >>= \case
```
Comment (ksaric) [inline app/daedalus-ipc.hs]: `fmap readEither <$>` == `readEither <<$>>`

File: src/Cardano/Shell/DaedalusIPC.hs [hunk 2]
```diff
@@ -80,154 +62,26 @@ daedalusIPC
...
     -> IO ()
-daedalusIPC trace port = withNodeChannel (pure . msg) action >>= \case
-    Right runServer -> do
-        logInfo trace "Daedalus IPC server starting"
-        runServer >>= \case
+daedalusIPC tr port = setupNodeChannel >>= \case
+    Right chan -> do
+        logInfo tr "Daedalus IPC server starting"
+        runNodeChannel (pure . msg) action chan >>= \case
             Left (NodeChannelFinished err) ->
```
Comment (ksaric) [inline src/Cardano/Shell/DaedalusIPC.hs]: You probably can chain this?

**Comment by CodiePP** [comment]
this needs some update to `nix` files.

**Comment by rvl** [comment]
OK I updated the nix files. Up to you to decide on `<<$>>` - I don't normally use it.

**Pull request merged**